### PR TITLE
feat: migrate to bzlmod

### PR DESCRIPTION
**Pull request by thesayyn**
Migrating to bzlmod, needs https://github.com/GoogleContainerTools/rules_distroless/pull/70 first.

### CODE REVIEW

**Comment by loosebazooka** [comment]
I think the node references are gone?

**Comment by thesayyn** [comment]
> I think the node references are gone?

no those went into `private/extensions/`

**Comment by loosebazooka** [comment]
oh I see, well maybe a name got mangled? : https://github.com/GoogleContainerTools/distroless/actions/runs/10311366649/job/28544819823?pr=1640#step:6:57

**Comment by thesayyn** [comment]
> oh I see, well maybe a name got mangled? : https://github.com/GoogleContainerTools/distroless/actions/runs/10311366649/job/28544819823?pr=1640#step:6:57

right, there was an issue in the extension, fixed now.

**Comment by thesayyn** [comment]
@loosebazooka this is ready now.

**Comment by thesayyn** [comment]
cc @alexeagle as he promised to review.

**Comment by loosebazooka** [comment]
oh shoot, I merged something in that probs needs a rebase+fix

File: private/extensions/node.bzl [hunk 1]
```diff
@@ -1,9 +1,99 @@
-"repositories for nodejs"
+"node"
```
Comment (loosebazooka) [inline private/extensions/node.bzl:1]: @omBratteng fyi, if you are updating node packages they are now in private/extensions/node.bzl

**Comment by vipulroxx** [comment]
Put me in Google if you want me to fix it. Please otherwise remove this
spam. This is a easy by the way.

File: private/extensions/node.bzl (hunk 1 above)
Comment (omBratteng) [reply private/extensions/node.bzl:1]: 👍

**Comment by loosebazooka** [comment]
@thesayyn needs a rebase?

**Comment by mortenmj** [comment]
I get an error if I attempt to reference this as a dependency in our repository:

> ERROR: in module dependency chain <root> -> distroless@_: include() directive found at @@distroless~//:MODULE.bazel:60:1, but it can only be used in the root module. Type 'bazel help mod' for syntax and help.

**Comment by loosebazooka** [comment]
> There are some assumptions in this implementation that make it hard to import this in a downstream module. Will that be supported later?

I think fundamentally we **dont** want anyone importing distroless as a downstream module. If you wish to build distroless style images -- rules_distroless is what you need? What exactly is your usecase? Are there pieces in here that need to move into rules_distroless?

**Comment by thesayyn** [comment]
> @thesayyn needs a rebase?

finally got around to rebase. it should be ready now.  i can't confirm `scripts/update_java_archives.sh` works because it fails on a darwin bash...

**Comment by loosebazooka** [comment]
I can update the script for Java later

File: scripts/update_java_archives.sh [hunk 2]
```diff
@@ -39,20 +39,13 @@ function generate_java_archives() {
...
-    "temurin21_jdk_ppc64le": "${plain_version}",
-}
-
-def repositories():
-    "java archives"
+java = use_extension("//private/extensions:java.bzl", "java")
 EOM
+  exit 1
+
+
```
Comment (loosebazooka) [inline scripts/update_java_archives.sh:48]: this looks like a debug exit that was left in here?

**Comment by loosebazooka** [comment]
looks like the image diffs (non-java) are mostly just related to: https://github.com/GoogleContainerTools/rules_distroless/pull/71

**Comment by loosebazooka** [comment]
java versions need to be sync'd, there was an update here: https://github.com/GoogleContainerTools/distroless/pull/1649

File: scripts/update_java_archives.sh (hunk 2 above)
Comment (thesayyn) [reply scripts/update_java_archives.sh]: oh yes...

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**